### ISSUE TITLE
movies_actor() returns all types of titles (not just movies)

### ISSUE THREAD

**Issue by nmpgaspar** [tboothman/imdbphp#49]
It would be awesome if we could have movies_actor() just for actual movies, and tvshows_actor() for instance for tvshows. Alternatively, a title_actor() method that would include a field indicating the type of the title (Movie, TvShow or TvEpisode).

I believe this is feasible, since by just looking at a person page (e.g. http://www.imdb.com/name/nm0933988/) we can deduce the type of each title that person had a role in. My point is that is does not seem to require further page request to each title entry in the Person filmography..

Could it possible for you to consider implementing this?

**Comment by tboothman** [comment]
This whole functionality is a little flawed, I'd be surprised if anyone really uses it.
I've added in the type field to the filmographies, but you've still got the limitation of only getting the filmography sections that have functions, the other ones don't feature.
I think probably the most generic and therefore most useful function i could make is something like this
`function filmography($category = "", $titleTypes = array())`
You'd just get everything by default, then limit it as you'd like to only acting, only directing etc and only for films.
I've added in a bit of a fudge to keep the types consistent with the ones you get for the title. For example in the filmography documentary films show up as 'Documentary' but on the title page they are a movie in the documentary genre.
